baseagent.py

```diff
@@ -116,19 +117,22 @@ class OutputQueue( CyclicBehaviour ):
         async def run( self ):
             pass
 
-
     class Stop( CyclicBehaviour ):
-        # TODO: Test and validate this
-        #       Also look up async def stop in APiHolon
-        async def run( self ):
-            msg = await self.receive( timeout=0.1 )
+        async def run(self):
+            msg = await self.receive( timeout=1 )
             if msg:
-                try:
-                    performative = msg.metadata[ "performative" ]
-                    if performative == 'request' and msg.sender == self.agent.holon and msg.content == 'stop':
-                        self.kill()
-                except KeyError:
-                    pass
+                if self.agent.verify( msg ):
+                    self.agent.say( '(StopAgent) Message verified, processing ...' )
+                    self.agent.say( '(StopAgent) Holon has scheduled us to stop. Stopping!' )
+
+                    metadata = deepcopy( self.agent.inform_msg_template )
+                    metadata[ 'status' ] = 'stopped'
+                    await self.agent.schedule_message( self.agent.holon, metadata=metadata )
+
+                    # or should we use super().stop()?
+                    self.kill()
```

Comment:
Would we prefer to invoke `self.kill()` or `super.stop()` to stop the agent?

Reply:
Good question, I'm not sure what the difference is. [1] says that stop is used for behaviours, but I am not sure if it is for agents as well. Anyway, it isn't crucial at the moment ;-)


[1] https://spade-mas.readthedocs.io/en/latest/usage.html